### PR TITLE
Add conceptual clouds

### PR DESCRIPTION
**Pull request by lkluft**

### CODE REVIEW

File: konrad/cloud.py [hunk 1]
```diff
@@ -536,6 +546,89 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, coupling='pressure', **kwargs)
 
 
+class ConceptualCloud(DirectInputCloud):
+    def __init__(self, atmosphere, cloud_top, depth, cloud_fraction,
+                 water_path=100e-3, particle_size=100., phase='ice',
+                 coupling='pressure'):
+        """Initialize a grey cloud."""
```
Comment (SallyDa) [inline konrad/cloud.py:553]: Could you please add the parameters to the docstring and say what units they should have (if applicable).

Comment (lkluft) [reply konrad/cloud.py]: Fair point! I forgot to add that after I finished prototyping

File: konrad/cloudoptics.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            ('num_shortwave_bands',), ssa.ravel(),
+        )
+
+        cld_optics['cloud_asymmetry_parameter'] = (
+            ('num_shortwave_bands',), asym.ravel(),
+        )
+
+        cld_optics['cloud_forward_scattering_fraction'] = (
+            ('num_shortwave_bands',), asym.ravel() ** 2
+        )
```
[the remark this answers is not in the document]
Comment (lkluft) [reply konrad/cloudoptics.py:102]: This assumes Henyey-Greenstein phase functions (e.g. [Joseph et al., 1976](https://doi.org/10.1175/1520-0469(1976)033%3C2452:TDEAFR%3E2.0.CO;2), their Eq. 5a).
I am not sure if this is what they are actually doing but I think the approach is often used.

**Comment by lkluft** [comment]
> Looks good; it's nice that you have the ECHAM cloud optics data too. One could do a comparison of the impact of cloud optics on RCE - I've heard the two schemes are quite different, so I guess it would have a strong effect. I don't know how interesting that is though!

This is indeed an interesting question! But I one would need to the cloud parameters for botch cloud optics differently to result in the same cloud radiative effect. And I can tell you that even tuning one of them is annoying as... ;)

File: konrad/cloud.py (hunk 1 above)
Comment (SallyDa) [reply konrad/cloud.py]: Thanks! I find it weird that cloud depth is in pressure coordinates, which could definitely lead to some strange behaviour if the cloud moves too much. But if that's what you're using and it works for the cases you are investigating, it's fine!

File: konrad/cloudoptics.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (SallyDa) [reply konrad/cloudoptics.py:102]: Okay I have no idea about this. Maybe it would be good to check the ECHAM code?

**Comment by SallyDa** [comment]
> > Looks good; it's nice that you have the ECHAM cloud optics data too. One could do a comparison of the impact of cloud optics on RCE - I've heard the two schemes are quite different, so I guess it would have a strong effect. I don't know how interesting that is though!
> 
> This is indeed an interesting question! But I one would need to the cloud parameters for botch cloud optics differently to result in the same cloud radiative effect. And I can tell you that even tuning one of them is annoying as... ;)

Haha, I'm also not volunteering :)

File: konrad/cloudoptics.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lkluft) [reply konrad/cloudoptics.py:102]: It is also equivalent to the [CLIMT defaults](https://www.youtube.com/watch?v=TB-uB5MqiqU) ;)

File: konrad/cloud.py (hunk 1 above)
Comment (lkluft) [reply konrad/cloud.py]: As far as I know, there is not obvious choice between fixed depth in height or pressure. I will just stick to the latter for simplicity. I guess it is one of the smaller problems when representing clouds in our 1D-RCE ;)
The only interesting simulation I would like to do at some point is fixing the cloud base temperature to turn-off that feedback.

**Pull request merged**